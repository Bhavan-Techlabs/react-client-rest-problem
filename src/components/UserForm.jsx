import React from "react";
import { useState } from "react";

const UserForm = ({ user }) => {
  const [formState, setFormState] = useState({ name: "", address: "" });

  const onUpdate = (e) => {
    e.preventDefault();

    const updateUser = async (id, data) => {
      return await fetch(
        `https://608b7a57737e470017b74d29.mockapi.io/api/v1/users/${id}`,
        {
          method: "PUT",
          headers: {
            "Content-Type": "application/json",
          },
          body: JSON.stringify(data),
        }
      );
    };

    const formToSubmit = { ...user, ...formState };
    updateUser(user.id, formToSubmit);
    alert("Updated!");
  };

  return (
    <div className="form-wrapper">
      <form>
        <p>
          <label>Name: </label>
          <input
            name="name"
            type="text"
            value={formState.name}
            onChange={(e) => {
              setFormState((s) => ({ ...s, name: e.target.value }));
            }}
          />
        </p>
        <p>
          <label>Address: </label>
          <input
            name="address"
            type="text"
            value={formState.address}
            onChange={(e) => {
              setFormState((s) => ({ ...s, address: e.target.value }));
            }}
          />
        </p>
        <button onClick={onUpdate}>Update User</button>
      </form>
    </div>
  );
};

export default UserForm;
