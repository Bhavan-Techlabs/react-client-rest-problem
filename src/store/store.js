import { configureStore } from "@reduxjs/toolkit";

export const store = configureStore({
  //TODO: Config the reducer
  reducer: {}
});
