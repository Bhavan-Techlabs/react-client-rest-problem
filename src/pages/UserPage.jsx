import React from "react";
import { useEffect, useState } from "react";
import { useSelector } from "react-redux";
import { useParams } from "react-router-dom";
import User from "../components/User";
import UserForm from "../components/UserForm";

const UserPage = () => {
  const [user, setUser] = useState();
  const { userId } = useParams();
  const users = useSelector((state) => state.user.users);

  useEffect(() => {
    if (userId && users) {
      const user = users.find((user) => user.id === userId);
      if (user) {
        setUser(user);
      }
    }
  }, [userId]);

  return (
    <>
      <div className="user-page-wrapper">
        <h1>User Page</h1>
        {user && <User user={user} />}
        {user && <UserForm user={user} />}
      </div>
    </>
  );
};

export default UserPage;
