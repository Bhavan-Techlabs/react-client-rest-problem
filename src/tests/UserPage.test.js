import React from "react";
import renderer from "react-test-renderer";
import UserForm from "../components/UserForm";
import UserPage from "../pages/UserPage";
import { Provider } from "react-redux";

import { store } from "../store/store";

it("UserPage component renders", () => {
  const component = renderer.create(
    <Provider store={store}>
      <UserPage />
    </Provider>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("UserForm component renders", () => {
  const component = renderer.create(<UserForm />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
