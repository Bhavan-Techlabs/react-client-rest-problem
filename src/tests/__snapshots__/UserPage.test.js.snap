// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserForm component renders 1`] = `
<div
  className="form-wrapper"
>
  <form>
    <p>
      <label>
        Name: 
      </label>
      <input
        name="name"
        onChange={[Function]}
        type="text"
        value=""
      />
    </p>
    <p>
      <label>
        Address: 
      </label>
      <input
        name="address"
        onChange={[Function]}
        type="text"
        value=""
      />
    </p>
    <button
      onClick={[Function]}
    >
      Update User
    </button>
  </form>
</div>
`;

exports[`UserPage component renders 1`] = `
<div
  className="user-page-wrapper"
>
  <h1>
    User Page
  </h1>
</div>
`;
